CMakeLists.txt

```diff
@@ -555,6 +578,32 @@ elseif(${ARCH} STREQUAL "x86_64")
    set(KERNEL_BASE_VA     0xFFFFFFFF80000000)
    set(KERNEL_VADDR       0xFFFFFFFF80100000)  # -2 GB + 1 MB
 
+elseif(${ARCH} STREQUAL "riscv64")
+
+   set(ARCH_SHORT       "riscv64")
+   set(ARCH_FAMILY      "riscv")
+   set(ARCH_ELF_NAME    "elf64-littleriscv")
+   set(ARCH_LD_OUTPUT   "elf64lriscv")
+   set(ARCH_BFD         "riscv:rv64")
+   set(ARCH_GCC_TC      "riscv64")
+   set(ARCH_GCC_FLAGS   "-march=rv64imafdc -mabi=lp64d -mcmodel=medany")
+   set(ARCH_GCC_FLAGS   "${ARCH_GCC_FLAGS} -D__riscv64")
+   set(ARCH_BITS        64)
+
+   # Fundamental kernel MM constants
+   set(BASE_VA                     0x2000000000)  # +128 GB
+   set(KERNEL_PADDR                  0x80200000)  # Default
+
+   if (KRN32_LIN_VADDR)
```

Comment:
NOTE: on 32-bit it makes sense to support KRN32_LIN_VADDR because it makes sense to emulate the 64-bit case where the kernel is **not** part of the linear virtual address space and requires a dedicated mapping. That's only because the 64-bit port is not ready yet. Do you wanna support both the modes for RISCV64?

Reply:
The KRN32_LIN_VADDR option is here just to test that everything works fine. One principle of riscv porting code is that riscv64 and riscv32 share the same code.

Reply:
On riscv64, both modes are supported.

Reply:
OK, got. Since you're thinking also about the 32-bit port, it makes sense to support (and test) both modes now.